build_wasm.sh

```diff
@@ -20,18 +17,21 @@ FUNCTIONS='['
 FUNCTIONS+='_dedx_get_program_list,'
 FUNCTIONS+='_dedx_get_material_list,'
 FUNCTIONS+='_dedx_get_ion_list,'
-FUNCTIONS+='_dedx_get_min_energy,'
-FUNCTIONS+='_dedx_get_max_energy,'
 FUNCTIONS+='_dedx_get_ion_name,'
 FUNCTIONS+='_dedx_get_material_name,'
 FUNCTIONS+='_dedx_get_program_name,'
+FUNCTIONS+='_dedx_get_min_energy,'
+FUNCTIONS+='_dedx_get_max_energy,'
+FUNCTIONS+='_dedx_get_simple_stp,'
 FUNCTIONS+='_malloc,'
 FUNCTIONS+='_free'
 
-
 FUNCTIONS+=']'
 
-emcc libdedx.a -o $PROJECT_NAME.js -s EXPORTED_FUNCTIONS="$FUNCTIONS" -s ENVIRONMENT='web' -s USE_ES6_IMPORT_META=0 -s EXPORT_ES6=1 -s MODULARIZE=1 -s WASM=1 -s EXPORTED_RUNTIME_METHODS=["ccall","cwrap","UTF8ToString"]
+# remember to use --embed-file instead of --preload-file for web based development
+emcc libdedx.a -o $PROJECT_NAME.js -s EXPORTED_FUNCTIONS="$FUNCTIONS" -s ENVIRONMENT='web' -s USE_ES6_IMPORT_META=0 -s EXPORT_ES6=1 -s MODULARIZE=1 -s WASM=1 -s EXPORTED_RUNTIME_METHODS=["ccall","cwrap","UTF8ToString"] --embed-file ../../libdedx/data@data/
```

Comment:
what is that path `../../libdedx/data@data/` ?
Why `@` inside ?

Does it mean mount `../../libdedx/data` inside `wasm` into `data` directory ?

if yes, then explain it in comments

Reply:
the @ specifies where in the filesystem created by emcc the files would reside. So in this instance it's `../../libdedx/data@data/` which translates as take files from directory `../../libdedx/data` and put them in directory data in the newly created filesystem. It's there since in libdedx the local directory path is data/ and so the dedx_file_access.c expects to find the files in the data dir.